experimental/sev_guest/src/interrupts.rs

```diff
@@ -0,0 +1,189 @@
+                "push r8",
+                "push r9",
+                "push r10",
+                "push r11",
+                // Make sure the stack is 16-byte aligned.
+                "sub rsp, 8",
+
+                // Back up the AVX registers.
+                // TODO(#3329): Update interrupt handler macro to support AVX, SSE or neither.
+                "sub rsp, 16*32",
+                "vmovups [rsp + 0*32], YMM0",
+                "vmovups [rsp + 1*32], YMM1",
+                "vmovups [rsp + 2*32], YMM2",
+                "vmovups [rsp + 3*32], YMM3",
+                "vmovups [rsp + 4*32], YMM4",
+                "vmovups [rsp + 5*32], YMM5",
+                "vmovups [rsp + 6*32], YMM6",
+                "vmovups [rsp + 7*32], YMM7",
+                "vmovups [rsp + 8*32], YMM8",
+                "vmovups [rsp + 9*32], YMM9",
+                "vmovups [rsp + 10*32], YMM10",
+                "vmovups [rsp + 11*32], YMM11",
+                "vmovups [rsp + 12*32], YMM12",
+                "vmovups [rsp + 13*32], YMM13",
+                "vmovups [rsp + 14*32], YMM14",
+                "vmovups [rsp + 15*32], YMM15",
+
+                // Call the inner function with the System V calling convention. Argument 1 is
+                // already in RDI and argument 2 in RSI.
+                "call  {INNER_ADDRESS}",
+
+                // Restore AVX registers.
+                "vmovups YMM0, [rsp + 0*32]",
+                "vmovups YMM1, [rsp + 1*32]",
+                "vmovups YMM2, [rsp + 2*32]",
+                "vmovups YMM3, [rsp + 3*32]",
+                "vmovups YMM4, [rsp + 4*32]",
+                "vmovups YMM5, [rsp + 5*32]",
+                "vmovups YMM6, [rsp + 6*32]",
+                "vmovups YMM7, [rsp + 7*32]",
+                "vmovups YMM8, [rsp + 8*32]",
+                "vmovups YMM9, [rsp + 9*32]",
+                "vmovups YMM10, [rsp + 10*32]",
+                "vmovups YMM11, [rsp + 11*32]",
+                "vmovups YMM12, [rsp + 12*32]",
+                "vmovups YMM13, [rsp + 13*32]",
+                "vmovups YMM14, [rsp + 14*32]",
+                "vmovups YMM15, [rsp + 15*32]",
+                "add rsp, 16*32",
+
+                // Undo stack alignment.
+                "add rsp, 8",
+                // Restore scratch registers.
+                "pop r11",
+                "pop r10",
+                "pop r9",
+                "pop r8",
+                // Restore potentially modified general-purpose registers from the interrupt stack
+                // frame.
+                "pop rax",
+                "pop rbx",
+                "pop rcx",
+                "pop rdx",
+                "pop rdi",
+                "pop rsi",
+                // The stack should now be in the original state minus the error code, so return
+                // from the handler.
+                "iretq",
+                INNER_ADDRESS = sym inner_function,
+                options(noreturn)
```

Comment:
minor nit, but I think we've used AT&T syntax everywhere else, so I think it would be good to maintain consistency and have this in AT&T and not Intel syntax as well. (Or translate everything else into Intel syntax, I'm also fine with that.)

Reply:
The rest of this crate also uses Intel syntax, although it would mostly look the same for most of the other cases in this crate.

I personally really dislike AT&T syntaxt, especially since I think something like `[rsp + 13*32]` is must more readable than something like `0x1a0(%rsp)`.

But I agree with the sentiment that we should be consistent. I have created #3335 to track moving to a consistent syntax.